packages/compass-assistant/src/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { registerCompassPlugin } from '@mongodb-js/compass-app-registry';
+import { AssistantProvider } from './assistant-provider';
+import { Chat } from './@ai-sdk/react/chat-react';
+import { docsProviderTransport } from './docs-provider-transport';
+
+const CompassAssistantProvider = registerCompassPlugin(
+  {
+    name: 'CompassAssistant',
+    component: AssistantProvider,
+    activate: () => {
+      const chat = new Chat({ transport: docsProviderTransport });
+      return {
+        store: { state: { chat } },
+        deactivate: () => {},
+      };
+    },
+  },
+  {
+    transport: () => docsProviderTransport,
+  }
```

Comment:
This needs to be a specially created service locator fuction if you want to inject it, but also doesn't look like you're actually using it? Not a bad idea though, will allow you to override it in tests for example